### PR TITLE
Backport master latest commits to 19.03

### PR DESCRIPTION
**Pull request by glours**
Backporting commits from `master` to be close as possible with the nightly docker-ce packaging in the `19.03` branch

backport of:

- https://github.com/docker/docker-ce-packaging/pull/357 Assorted improvements to Makefiles
- https://github.com/docker/docker-ce-packaging/pull/374 Dockerfiles: remove unused GO_VERSION
- https://github.com/docker/docker-ce-packaging/pull/390 plugins: remove (now) redundant .gitkeep
- https://github.com/docker/docker-ce-packaging/pull/395 Update packages list
- https://github.com/docker/docker-ce-packaging/pull/396 [TAR-1174][master] Turn Go11Module=off
- https://github.com/docker/docker-ce-packaging/pull/403 Dockerfiles: re-use DISTRO, SUITE build-args where possible
- https://github.com/docker/docker-ce-packaging/pull/404 deb: Only recommend aufs-tools on amd64
- https://github.com/docker/docker-ce-packaging/pull/411 Remove image-based build (for docker engine activate)
- https://github.com/docker/docker-ce-packaging/pull/420 Remove some remnants related to image-based builds
- https://github.com/docker/docker-ce-packaging/pull/421 Makefile: enable buildkit when building images
- https://github.com/docker/docker-ce-packaging/pull/426 Remove more remnants from image-based engine
- https://github.com/docker/docker-ce-packaging/pull/427 update list of reviewers in unir config
- https://github.com/docker/docker-ce-packaging/pull/462 Makefile: chown engine dir during cleanup

[TAR-1174]: https://docker.atlassian.net/browse/TAR-1174

### CODE REVIEW

File: deb/Makefile [hunk 1]
```diff
@@ -47,11 +48,6 @@ clean: ## remove build artifacts
 	$(RM) -r debbuild
 	[ ! -d sources ] || $(CHOWN) -R $(shell id -u):$(shell id -g) sources
 	$(RM) -r sources
-	$(RM) engine-image
```
Comment (thaJeztah) [inline deb/Makefile]: We need to discuss if we can already backport these changes

**Comment by thaJeztah** [comment]
Can you sign-off the cherry-pick commits? Some checks expect the cherry-pick commits to have a sign-off (which should be the last line in the commit); when cherry-picking, the `cherry-picked from` line is added, causing those checks to fail;

```
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
(cherry picked from commit 2350856)
```

You can amend the existing commits, or use both `-x` and `-s` when cherry-picking

File: deb/Makefile (hunk 1 above)
Comment (glours) [reply deb/Makefile]: Sure, that the main objective of this PR, what should we pick up to be close as possible of the nightly build without breaking the existing release process for previous version.

**Comment by thaJeztah** [comment]
Given that there's some bigger changes in this (removal of publishing the "engine" images); I suggest holding off merging this until after 19.03.7 has been done.

(I'm ok with a separate PR with a more limited number of changes if there's fixes in this one that are trivial and/or important to have)

**Comment by thaJeztah** [comment]
hmm; why is static failing on this branch, but not on master?

```
+ pwd
+ make ENGINE_DIR=/home/ubuntu/workspace/docker-ce-packaging_PR-433/engine clean-engine
rm -rf /home/ubuntu/workspace/docker-ce-packaging_PR-433/engine
rm: cannot remove '/home/ubuntu/workspace/docker-ce-packaging_PR-433/engine/bundles/binary-daemon/vpnkit': Permission denied
rm: cannot remove '/home/ubuntu/workspace/docker-ce-packaging_PR-433/engine/bundles/binary-daemon/rootlesskit-docker-proxy': Permission denied
rm: cannot remove '/home/ubuntu/workspace/docker-ce-packaging_PR-433/engine/bundles/binary-daemon/ctr.sha256': Permission denied
```

**Comment by tiborvass** [comment]
If only we had these things called containers so we could reproduce CI issues locally... #NotAFanOfJenkins

**Comment by tiborvass** [comment]
I think what's happening is we chown the workspace to jenkins user, but then the generated binaries are not chowned as jenkins user (probably root) so it can't remove it.

**Comment by thaJeztah** [comment]
Yes, there's some "chown-ing" happening in some parts; mostly curious why it fails in this branch but not on master

**Comment by thaJeztah** [comment]
diff with master; https://github.com/glours/docker-ce-packaging/compare/backport-master-to-19.03..docker:master

**Comment by tiborvass** [comment]
@thaJeztah is it normal that CI is using docker/engine ? I lost track of where we are.

**Comment by thaJeztah** [comment]
> is it normal that CI is using docker/engine ? I lost track of where we are with that.

Currently, yes, but I have a PR to change that; https://github.com/docker/docker-ce-packaging/pull/457 (let me extract the first commit to a separate PR for easier review/merge)

**Comment by tiborvass** [comment]
I simply rebased

**Comment by thaJeztah** [comment]
opened https://github.com/docker/docker-ce-packaging/pull/462 and cherry-picked it here to see if that resolves the problem

**Comment by thaJeztah** [comment]
All green now, with https://github.com/docker/docker-ce-packaging/pull/462 cherry-picked (we should merge that one first though)

**Pull request merged**